### PR TITLE
alternative fifo

### PR DESCRIPTION
**Pull request by ghost**
Doesn't this essentially do the same thing?

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -14,15 +14,59 @@ how to use the page table and disk interfaces.
...
+	    frame = f;
+	    break;
+	}
+    }
+    if(frame == -1){
+        // Pick a random page that owns a frame.
+        int curr_page;
+       
+	curr_page = curr_frame % pt->npages;
+
```
Comment (ghost) [inline main.c]: This will always evict the oldest page

Comment (elliott-beach) [reply main.c]: I just realized that what I implemented was a LIFO queue. This is a good solution.

File: main.c [hunk 2]
```diff
@@ -50,7 +94,7 @@ int main( int argc, char *argv[] )
 	char *physmem = page_table_get_physmem(pt);
 
 	if(!strcmp(program,"sort")) {
-		sort_program(virtmem,npages*PAGE_SIZE);
+		sort_program(virtmem, npages * 5);
```
Comment (elliott-beach) [inline main.c]: I had the page count set to 5 hear in an earlier commit, but I would set it to PAGE_SIZE when testing so that you know it works.

Comment (elliott-beach) [reply main.c]: Also, I was testing the paging in the other PRs by printing, in `program.c`, the sum of sorting before and after running the quick-sort. Like https://github.com/elliott-beach/project2_base/blob/59c471a75a3fd18f3acd5385c266b4d527bde8d1/program.c#L63-L82

The sum will be different if the paging corrupts the memory.
Could you test things that way?

Comment (elliott-beach) [reply main.c]: I think `cur_page++` should go after page is evicted,  not after every page is added.

 To be **first-in => first out** ,  @`curr_page` shouldn't be incremented until the page is evicted. That way page 0 is evicted once we run out of frames, then page 1, etc. Current code always evicts the last frame to be used.

Even with that change, this doesn't pass that test I suggested, not sure why.

Comment (ghost) [reply main.c]: I am assuming you are referring to `cur_frame` instead of `curr_page`? If so, I can make that change.

**Comment by ghost** [comment]
I removed some code from the existing function and made a few changes to pass the test.

I think before I was cycling through the pages in choosing what to evict, when we need to be cycling through the frames and evict the page that is in that frame.  Now it selects a frame, determines which page is in it, and evicts that page

**Comment by elliott-beach** [comment]
I want to look at how this cycles through pages during the quick-sort operation.

**Comment by elliott-beach** [comment]
I think that this works, although I don't quite understand why.

**Comment by elliott-beach** [comment]
Thanks for this @kstowe. Now we just need to do the custom paging algorithm, the experiment, and the write-up - EZ.

**Comment by elliott-beach** [comment]
Merged.
I also added a check in `main` that uses this if "fifo" is the command-line arg for the algorithm.

**Comment by elliott-beach** [comment]
I also added a check in `main` that uses this if "fifo" is the command-line arg for the algorithm.

**Pull request closed without merging**